src/mlpack/methods/ann/rbm/binary_rbm_policy_impl.hpp

```diff
@@ -0,0 +1,159 @@
+    ::Evaluate(DataType& predictors, size_t i)
+{
+  size_t idx = RandInt(0, predictors.n_rows);
+  DataType temp = arma::round(predictors.col(i));
+  corruptInput = temp;
+  corruptInput.row(idx) = 1 - corruptInput.row(idx);
+  return std::log(LogisticFunction::Fn(FreeEnergy(std::move(corruptInput)) -
+      FreeEnergy(std::move(temp)))) * predictors.n_rows;
+}
+
+template<typename DataType>
+void BinaryRBMPolicy<DataType>
+    ::PositivePhase(DataType&& input, DataType&& gradient)
+{
+  DataType weightGrad = DataType(gradient.memptr(),
+      hiddenSize, visibleSize, false, false);
+
+  DataType hiddenBiasGrad = DataType(gradient.memptr() + weightGrad.n_elem,
+      hiddenSize, 1, false, false);
+
+  DataType visibleBiasGrad = DataType(gradient.memptr() + weightGrad.n_elem +
+      hiddenBiasGrad.n_elem, visibleSize, 1, false, false);
+
+  HiddenMean(std::move(input), std::move(hiddenBiasGrad));
+  weightGrad = hiddenBiasGrad * input.t();
+  visibleBiasGrad = input;
+}
+
+template<typename DataType>
+void BinaryRBMPolicy<DataType>
+    ::NegativePhase(DataType&& negativeSamples, DataType&& gradient)
+{
+  DataType weightGrad = DataType(gradient.memptr(),
+      hiddenSize, visibleSize, false, false);
+
+  DataType hiddenBiasGrad = DataType(gradient.memptr() + weightGrad.n_elem,
+      hiddenSize, 1, false, false);
+
+  DataType visibleBiasGrad = DataType(gradient.memptr() + weightGrad.n_elem +
+      hiddenBiasGrad.n_elem, visibleSize, 1, false, false);
+
+  HiddenMean(std::move(negativeSamples), std::move(hiddenBiasGrad));
+  weightGrad = hiddenBiasGrad * negativeSamples.t();
+  visibleBiasGrad = negativeSamples;
+}
+
+template<typename DataType>
+void BinaryRBMPolicy<DataType>
+    ::VisibleMean(DataType&& input, DataType&& output)
+{
+  VisiblePreActivation(std::move(input), std::move(output));
+  LogisticFunction::Fn(output, output);
+}
+
+template<typename DataType>
+void BinaryRBMPolicy<DataType>
+    ::HiddenMean(DataType&& input, DataType&& output)
+{
+  HiddenPreActivation(std::move(input), std::move(output));
+  LogisticFunction::Fn(output, output);
+}
+
+template<typename DataType>
+void BinaryRBMPolicy<DataType>
+    ::SampleVisible(DataType&& input, DataType&& output)
+{
+  VisibleMean(std::move(input), std::move(output));
+
+  for (size_t i = 0; i < output.n_elem; i++)
+    output(i) = math::RandBernoulli(output(i));
+}
+
+template<typename DataType>
+void BinaryRBMPolicy<DataType>
+    ::SampleHidden(DataType&& input, DataType&& output)
+{
+  HiddenMean(std::move(input), std::move(output));
+
+  for (size_t i = 0; i < output.n_elem; i++)
+    output(i) = math::RandBernoulli(output(i));
+}
+
+template<typename DataType>
+void BinaryRBMPolicy<DataType>
+    ::VisiblePreActivation(DataType&& input, DataType&& output)
+{
+  output = weight.t() * input + visibleBias;
+}
+
+template<typename DataType>
+void BinaryRBMPolicy<DataType>
+    ::HiddenPreActivation(DataType&& input, DataType&& output)
```

Comment:
Maybe I am mistaken, but it seems the style guide suggested to write code without line-breaks between
class names and `::`.